server/src/test/java/org/elasticsearch/cluster/routing/allocation/decider/DiskThresholdDeciderUnitTests.java

```diff
@@ -471,16 +473,95 @@ public void testShardSizeAndRelocatingSize() {
             other_0.getTargetRelocatingShard()
         );
         if (other_0.primary()) {
-            assertEquals(10100L, sizeOfRelocatingShards(allocation, node, false, "/dev/null"));
-            assertEquals(10090L, sizeOfRelocatingShards(allocation, node, true, "/dev/null"));
+            assertEquals(10100L, sizeOfUnaccountedShards(allocation, node, false, "/dev/null"));
+            assertEquals(10090L, sizeOfUnaccountedShards(allocation, node, true, "/dev/null"));
         } else {
-            assertEquals(100L, sizeOfRelocatingShards(allocation, node, false, "/dev/null"));
-            assertEquals(90L, sizeOfRelocatingShards(allocation, node, true, "/dev/null"));
+            assertEquals(100L, sizeOfUnaccountedShards(allocation, node, false, "/dev/null"));
+            assertEquals(90L, sizeOfUnaccountedShards(allocation, node, true, "/dev/null"));
         }
     }
 
-    public long sizeOfRelocatingShards(RoutingAllocation allocation, RoutingNode node, boolean subtractShardsMovingAway, String dataPath) {
-        return DiskThresholdDecider.sizeOfRelocatingShards(
+    public void testTakesIntoAccountExpectedSizeForInitializingSearchableSnapshots() {
+        Metadata metadata = Metadata.builder()
+            .put(
+                IndexMetadata.builder("test")
+                    .settings(
+                        settings(Version.CURRENT).put("index.uuid", "1234")
+                            .put(INDEX_STORE_TYPE_SETTING.getKey(), SEARCHABLE_SNAPSHOT_STORE_TYPE)
+                    )
+                    .numberOfShards(3)
+                    .numberOfReplicas(1)
+            )
+            .build();
+
+        Index index = new Index("test", "1234");
+        String nodeId = "node1";
+        ShardRouting shard1 = ShardRoutingHelper.moveToStarted(
```

Comment:
Can we instead create the 3 shards in a loop? Seems overly verbose spelled out here.

Reply:
Not sure I'm following. I believe shards don't get filtered by the node id. `RoutingNode` just index all the shards that are supplied in its constructor.